Mobile optimization

chrismaree opened the pull request:
This PR makes the dapp scale dynamically with different screen sizes. The screenshots below showcase some of the changes. Plz play with it on a staging branch to fully see all changes.

Resolves #41 

![image](https://user-images.githubusercontent.com/12886084/88487990-7eeb2e80-cf8a-11ea-9ce1-e058ccab85f7.png)

![image](https://user-images.githubusercontent.com/12886084/88487998-8579a600-cf8a-11ea-8477-b5f377facf56.png)

Very small screen:
![image](https://user-images.githubusercontent.com/12886084/88488004-93c7c200-cf8a-11ea-99df-e89180fa7104.png)

Create on Iphone X
![image](https://user-images.githubusercontent.com/12886084/88488014-a3dfa180-cf8a-11ea-8a83-597327d6c774.png)

Wrap unwrap
![image](https://user-images.githubusercontent.com/12886084/88488017-ab06af80-cf8a-11ea-8cb8-f688e8b7a161.png)

Yield calculator:
![image](https://user-images.githubusercontent.com/12886084/88488020-b4901780-cf8a-11ea-868e-f962bfe2e250.png)

Footer:
![image](https://user-images.githubusercontent.com/12886084/88488071-1fd9e980-cf8b-11ea-9ba0-70a2c4d29efc.png)

nicholaspai commented on features/emp-selector/EmpSelector.tsx:
```diff
@@ -43,6 +48,10 @@ const EmpSelector = () => {
     setEmpAddress(value === 0 ? null : (value as string));
   };
 
+  const prettyAddress = (x: string) => {
```
Nice +1

nicholaspai commented on features/manage-position/YourPosition.tsx:
```diff
@@ -121,37 +121,40 @@ const YourPosition = () => {
         </Status>
         <Status>
-          <Label>Tokens outstanding: </Label>
+          <Label>Tokens created: </Label>
```
Why switch this to "created"? I feel like "outstanding" is more intuitive.

chrismaree replied:
mainly was the width of the text.

chrismaree replied:
I have reverted it to your original text

nicholaspai commented on features/manage-position/YourPosition.tsx:
```diff
@@ -121,37 +121,40 @@ const YourPosition = () => {
             ):{" "}
           </Label>
-          {_latestPrice}
+          {Number(_latestPrice).toFixed(4)}
```
`_latestPrice` is already passed in as `latestPrice.toFixed(4)`

chrismaree replied:
it wasn't though. I added this in to make it scaled correctly. there was perhaps a change in between my PR most recently.

chrismaree replied:
not that I can see. looks like this without the toFixed(4)
![image](https://user-images.githubusercontent.com/12886084/88549474-9e349b00-d020-11ea-8bde-13bd2620000f.png)

nicholaspai replied:
No look at line 89 where this is passed in. It is set to 6 decimals. This was originally set to 6 from 4 because 4 was not enough for ETHBTC

nicholaspai replied:
Also can you re-add the space back between "price" and "("

nicholaspai commented on features/manage-position/YourPosition.tsx:
```diff
@@ -121,37 +121,40 @@ const YourPosition = () => {
         <Status>
-          <Label>(CR) Collateralization ratio:</Label>
-          {` ${pricedCR} (${_collSymbol} / ${_tokenSymbol})`}
+          <Label>Collateralization ratio(CR):</Label>
```
There should be a space between "ratio" and "(CR)"

chrismaree replied:
I removed this intentionally to save space for mobile screen sizes but can add it back.

nicholaspai replied:
I'd like to add it back they shouldn't be in the same word (without spaces between)

nicholaspai commented on features/manage-position/YourPosition.tsx:
```diff
@@ -121,37 +121,40 @@ const YourPosition = () => {
         <Status>
-          <Label>(GCR) Global collateralization ratio:</Label>
-          {` ${pricedGCR} (${_collSymbol} / ${_tokenSymbol})`}
+          <Label>Global collateralization ratio(GCR):</Label>
```
Same as above

chrismaree replied:
here is an example of this string on mobile:
![image](https://user-images.githubusercontent.com/12886084/88556233-12733c80-d029-11ea-9379-62c8ffd29299.png)

chrismaree replied:
and with a EMP selected looks like this:
![image](https://user-images.githubusercontent.com/12886084/88556423-4babac80-d029-11ea-94d3-217147a6041a.png)

nicholaspai replied:
In general I don't think we should be fixing mobile optimization by removing spaces. We should toggle font-size or the text in general for "small" screens instead

nicholaspai commented on features/manage-position/YourPosition.tsx:
```diff
@@ -121,37 +121,40 @@ const YourPosition = () => {
-          <Label>Liquidation Price:</Label>
-          {` ${liquidationPrice} (${priceIdUtf8})`}
+          <Label>Liquidation price:</Label>
+          {` ${liquidationPrice}(${priceIdUtf8})`}
```
Why do you prefer to not have spaces between values and their symbols?

chrismaree replied:
mainly wanted the width to be narrow.

chrismaree replied:
main was trying to save as much width as posible

nicholaspai commented on features/yield-calculator/YieldCalculator.tsx:
```diff
@@ -111,14 +111,14 @@ const YieldCalculator = () => {
                   variant="outlined"
                   inputProps={{ min: "0", max: "10", step: "0.01" }}
-                  helperText={`Enter the market observable price`}
+                  helperText={`Enter the market price`}
```
I wanted to change this to "Enter the price of yUSD of $" as it is even more explicit what to do

nicholaspai commented on features/yield-calculator/YieldCalculator.tsx:
```diff
@@ -139,12 +139,21 @@ const YieldCalculator = () => {
+                  </Typography>
+                </Box>
+              </Hidden>
+              <Hidden only={["md", "lg", "xl"]}>
```
What's the point of `Hidden`?

chrismaree replied:
depending on the screen size there needs to be different padding. `<Hidden only={["md", "lg", "xl"]}>` will hide the code block on `md` `lg` and `xl` screen sizes.

nicholaspai commented on features/emp-selector/EmpSelector.tsx:
```diff
@@ -1,6 +1,8 @@
 import styled from "styled-components";
-import Box from "@material-ui/core/Box";
+import { Box, Hidden } from "@material-ui/core";
```
Do you use `Hidden` ?

nicholaspai commented on pages/index.tsx:
```diff
@@ -27,18 +38,81 @@ const Blurb = styled.div`
+              <Grid container spacing={2}>
+                <Grid item>
+                  <Button variant="contained" onClick={handleClickListItem}>
+                    Menu
```
Thoughts on changing this to an IconButton using the "menu" icon from https://material-ui.com/components/material-icons/ ?

chrismaree replied:
ye, I'll add a burger :)

chrismaree replied:
done 
![image](https://user-images.githubusercontent.com/12886084/88548891-f3bc7800-d01f-11ea-821f-ba464f3ff963.png)

nicholaspai replied:
You like this one over Menu more?

chrismaree commented:
> Looks great overall thanks for adding mobile optimization. I'm a big fan of the
> 
> button for smaller screens.
> However, I disagree with some of your text/copy especially having no spaces between values and symbols. Please check out comments!

Sweet, thanks for the review! I've added back the spacing and reverted the wording according to the comments.

nicholaspai commented on features/contract-state/GeneralInfo.tsx:
```diff
@@ -139,7 +139,7 @@ const GeneralInfo = () => {
 
         <Status>
-          <Label>GCR (collateral / tokens): </Label>
+          <Label>Global Collateral ratio: </Label>
```
Capitalize the "R" in ratio, or lowercase the "C" in collateral to be consistent.

chrismaree replied:
ye, you right. this needs to be consistent. have updated it

nicholaspai commented on features/manage-position/Create.tsx:
```diff
@@ -235,44 +235,47 @@ const Create = () => {
+                    fullWidth
+                    variant="contained"
+                    onClick={mintTokens}
+                    disabled={
```
If you're only going to show this button when they don't need allowance, can you remove the `needAllowance` from the disabled={} prop since its redundant now?

nicholaspai commented on features/manage-position/Deposit.tsx:
```diff
@@ -144,28 +144,32 @@ const Deposit = () => {
+                  variant="contained"
+                  onClick={depositCollateral}
+                  disabled={
+                    needAllowance ||
```
Same comment as above for Create

nicholaspai commented on features/manage-position/MethodSelector.tsx:
```diff
@@ -59,19 +59,19 @@ const MethodSelector = ({ method, handleChange }: IProps) => {
             <ListItemText
               primary="Deposit"
-              secondary="Deposit additional collateral."
+              secondary="Top up your position's collateral."
```
Is "Top up your position's" preferable to "Deposit additional"? I find the former more precise language and less casual. Depends what vibe we're going for I guess

nicholaspai replied:
Oh well, I guess this is `secondary` text. I recommend changing "Top up" with "Add to" to parallel the "Remove" language. Also remove "your".

Recommendation: "Add to position collateral"

chrismaree replied:
ye, I changed this because the menu was so repetitive. All the actions said basically the same thing as the primary:
![image](https://user-images.githubusercontent.com/12886084/88572113-efec1e00-d03e-11ea-8362-dbcbaf173a83.png)

Will change to your recommendation.

nicholaspai commented on features/manage-position/MethodSelector.tsx:
```diff
@@ -59,19 +59,19 @@ const MethodSelector = ({ method, handleChange }: IProps) => {
             <ListItemText
               primary="Withdraw"
-              secondary="Withdraw excess collateral."
+              secondary="Remove excess collateral."
```
Recommendation: "Remove position collateral"

nicholaspai commented on features/manage-position/YourPosition.tsx:
```diff
@@ -115,49 +115,64 @@ const YourPosition = () => {
+          <Grid item md={6} xs={12}>
+            <Typography variant="h5">Your Position</Typography>
+            <Status>
+              <Label>Collateral supplied: </Label>
```
These <Labels> have a space at the end and no space preceding the string that follows.

I recommend changing all of the Labels to have a trailing space and remove the preceding space in the following text. So for example, change:

```
<Label>Collateral ratio (CR):</Label>
{` ${pricedCR}`}
```

to 

```
<Label>Collateral ratio (CR): </Label>
{`${pricedCR}`}
```

chrismaree replied:
Good catch. looks like this now:
![image](https://user-images.githubusercontent.com/12886084/88572308-3c375e00-d03f-11ea-9da4-2b3a96049520.png)

nicholaspai commented on features/manage-position/YourPosition.tsx:
```diff
@@ -115,49 +115,64 @@ const YourPosition = () => {
+                </Link>
+                ):{" "}
+              </Label>
+              {Number(_latestPrice).toFixed(4)}
```
If you think 4 decimals is better than 6, just change the `_latestPrice` that is passed into `renderComponent` instead of manipulating down here. I really want to keep our TSX components as stateless as possible

chrismaree replied:
I have decreased text size on `xs` screens which helped a lot.

nicholaspai commented on features/manage-position/YourPosition.tsx:
```diff
@@ -115,49 +115,64 @@ const YourPosition = () => {
+
+            <Status>
+              <Label>Global collateral ratio (GCR):</Label>
+              {` ${pricedGCR}`}
```
Same comments as above about where to put the space

nicholaspai replied:
Still need to remove this

nicholaspai commented on features/contract-state/GeneralInfo.tsx:
```diff
@@ -108,20 +110,20 @@ const GeneralInfo = () => {
           <Label>Expiry Date: </Label>
           <Tooltip title={`Timestamp: ${expiryTimestamp}`} interactive>
-            <span>{expiryDate}</span>
+            <span>{expiryDate} UTC</span>
```
+1

nicholaspai commented on features/yield-calculator/YieldCalculator.tsx:
```diff
@@ -139,12 +139,21 @@ const YieldCalculator = () => {
+              <Hidden only={["sm", "xs"]}>
+                <Box pt={4}>
+                  <Typography variant="h6">
+                    Yearly APR: {prettyPercentage(yieldAmount)}%
```
You need to merge in the latest `develop` changes that affects this YieldCalculator.tsx file

nicholaspai commented on features/emp-selector/EmpSelector.tsx:
```diff
@@ -1,6 +1,8 @@
-import Box from "@material-ui/core/Box";
+import { Box } from "@material-ui/core";
 import { withStyles } from "@material-ui/core/styles";
+import useMediaQuery from "@material-ui/core/useMediaQuery";
```
nit: can't you import `useMediaQuery` in same struct as `Box`?

chrismaree replied:
ye, you right. will go through all the components and look for repeated imports. I suspect there are more than just this one.

nicholaspai commented on features/emp-selector/EmpSelector.tsx:
```diff
@@ -75,7 +84,12 @@ const EmpSelector = () => {
+                <ListItemText
+                  primary={largeScreen ? emp.name : emp.symbol}
+                  secondary={
+                    largeScreen ? emp.address : prettyAddress(emp.address)
```
Thiis is a nice trick

nicholaspai commented on features/manage-position/Create.tsx:
```diff
@@ -127,7 +127,8 @@ const Create = () => {
       collAllowance !== "Infinity" && collAllowance < collateralToDeposit;
-    const resultantTokensBelowMin = resultantTokens < minSponsorTokensFromWei;
+    const resultantTokensBelowMin =
+      resultantTokens < minSponsorTokensFromWei && resultantTokens != 0;
```
This is a good add but nit: add a second "=" . Its consistent with the rest of the code and we should avoid implicitly cast any types, which != does.

https://howtodoinjava.com/javascript/javascript-equality-vs-identity-operators/

nicholaspai commented on features/manage-position/Redeem.tsx:
```diff
@@ -175,37 +175,41 @@ const Redeem = () => {
+                  fullWidth
+                  variant="contained"
+                  disabled={
+                    needAllowance ||
```
Can you remove this `needAllowance`

nicholaspai commented on features/manage-position/YourPosition.tsx:
```diff
@@ -115,49 +115,64 @@ const YourPosition = () => {
+            </Status>
+            <Status>
+              <Label>Collateral ratio (CR): </Label>
+              {` ${pricedCR}`}
```
nit: remove this leading " "

nicholaspai commented on features/manage-position/YourPosition.tsx:
```diff
@@ -115,49 +115,64 @@ const YourPosition = () => {
+            </Status>
+            <Status>
+              <Label>Liquidation price: </Label>
+              {` ${liquidationPrice} (${priceIdUtf8})`}
```
Same here

nicholaspai commented on features/manage-position/YourPosition.tsx:
```diff
@@ -115,49 +115,64 @@ const YourPosition = () => {
+
+            <Status>
+              <Label>Pending withdrawal request: </Label>
+              {` ${_pendingWithdraw}`}
```
And here

nicholaspai commented on features/manage-position/YourPosition.tsx:
```diff
@@ -115,49 +115,64 @@ const YourPosition = () => {
+            {_pendingWithdraw == "available" && (
+              <Status>
+                <Label>Collateral available to withdraw: </Label>
+                {` ${_withdrawAmt}`}
```
here

nicholaspai commented on features/manage-position/YourPosition.tsx:
```diff
@@ -115,49 +115,64 @@ const YourPosition = () => {
+            </Status>
+            <Status>
+              <Label>Collateral requirement: </Label>
+              {` ${collReqFromWei}`}
```
Last one here
